--- enigma-1.01/src/d_models.cc	2006-08-15 12:49:10.000000000 -0700
+++ enigma-1.01/src/d_models.cc.new	2008-12-20 12:38:11.000000000 -0800
@@ -204,7 +204,7 @@
     if (lua::DoSysFile(L, "compat.lua") != lua::NO_LUAERROR) {
         std::string message = ecl::strf("Error loading 'compat.lua'\nError: '%s'\n",
                 lua::LastError(L).c_str());
-        fprintf(stderr, message.c_str());
+        fprintf(stderr, "%s", message.c_str());
         gui::ErrorMenu m(message + 
                 _("\n\nThis error may cause the application to behave strange!")
                 , N_("Continue"));
@@ -218,7 +218,7 @@
     if (lua::DoSysFile(L, vminfo->initscript) != lua::NO_LUAERROR) {
         std::string message = ecl::strf("Error loading '%s'\nError: '%s'\n",
                 fname.c_str(), lua::LastError(L).c_str());
-        fprintf(stderr, message.c_str());
+        fprintf(stderr, "%s", message.c_str());
         gui::ErrorMenu m(message + 
                 _("\n\nThis error may cause the application to behave strange!")
                 , N_("Continue"));
--- enigma-1.01/src/editor.cc	2006-07-29 09:41:44.000000000 -0700
+++ enigma-1.01/src/editor.cc.new	2008-12-20 12:40:21.000000000 -0800
@@ -211,11 +211,11 @@
     // TODO - just printing a message is not enough - the app will crash on missing editor.lua
     if (lua::DoSysFile(m_lua, "compat.lua") != lua::NO_LUAERROR) {
         std::string message = "While processing 'compat.lua':\n" +lua::LastError(m_lua);
-        fprintf( stderr, message.c_str() );
+        fprintf( stderr, "%s", message.c_str() );
     }
     if (lua::DoSysFile(m_lua, "editor.lua") != lua::NO_LUAERROR) {
         std::string message = "Error loading 'editor.lua'\n" +lua::LastError(m_lua);
-        fprintf( stderr, message.c_str() );
+        fprintf( stderr, "%s", message.c_str() );
     }
 }
 
--- enigma-1.01/src/main.cc	2007-05-25 14:28:44.000000000 -0700
+++ enigma-1.01/src/main.cc.new	2008-12-20 12:44:44.000000000 -0800
@@ -275,7 +275,7 @@
     // initialize preferences -- needs LUA, XML
     if (!options::Load()) {
         fprintf(stderr, _("Error in configuration file.\n"));
-      	fprintf(stderr, lua::LastError (lua::GlobalState()).c_str());
+      	fprintf(stderr, "%s", lua::LastError (lua::GlobalState()).c_str());
     }     
     prefs = PreferenceManager::instance();
     
